tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -0,0 +1,228 @@
+"""
+Name: post_copy_request_to_queue.py
+Description:  lambda function that queries the db for file metadata, updates the status
+of recovered file to staged, and sends the staged file info to staged_recovery queue for further processing.
+
+"""
+from enum import Enum
+import json
+import os
+import boto3
+from typing import Dict, Any, List, Optional
+from datetime import datetime, timezone
+import time
+import random
+from shared_recovery import (
+    OrcaStatus,
+    RequestMethod,
+    post_status_for_file_to_queue,
+    post_entry_to_queue,
+)
+import requests_db
+import database
+from cumulus_logger import CumulusLogger
+import logging
+
+# instantiate CumulusLogger
+LOGGER = CumulusLogger()
+
+
+def task(
+    records: Dict[str, Any],
+    db_queue_url: str,
+    recovery_queue_url: str,
+    max_retries: int,
+    retry_sleep_secs: int,
+    retry_backoff: int,
+) -> None:
+    """
+    Task called by the handler to perform the work.
+
+    This task queries all entries from orca_recoverfile table
+    that match the given filename and whose status_id is 'PENDING'.
+    The result is then sent to the staged-recovery-queue SQS and status-update-queue SQS.
+
+    Args:
+        records: A dictionary passed through from the handler.
+        db_queue_url: The SQS URL of status_update_queue
+        recovery_queue_url: The SQS URL of staged_recovery_queue
+        max_retries: Number of times the code will retry in case of failure.
+        retry_sleep_secs: Number of seconds to wait between recovery failure retries.
+        retry_backoff: The multiplier by which the retry interval increases during each attempt.
+    Returns:
+        None
+    Raises:
+        Exception: If unable to retrieve key_path or db parameters, convert db result to json,
+        or post to queue.
+
+    """
+    # grab the key_path and bucketname from event
+    # We only expect one record.
+    for record in records:
+        key_path = record["s3"]["object"]["key"]
+        bucket_name = record["s3"]["bucket"]["name"]
+
+    sql = """
+        SELECT
+            job_id, granule_id, filename, restore_destination
+        FROM
+            orca_recoverfile
+        WHERE
+            key_path = %s
+        AND
+            status_id = %d
+    """
+    # Gets the dbconnection info.
+    # query the db table
+    try:
+        db_connect_info = requests_db.get_dbconnect_info()
+        rows = database.single_query(sql, db_connect_info, (key_path, OrcaStatus.PENDING.value))
+        if len(rows) == 0:
+            message = f"No metadata found for {key_path}"
+            LOGGER.fatal(message)
+            raise Exception(message)
+        # # convert db result to json
+        db_result_json = database.result_to_json(rows)
+
+    except Exception as ex:
+        message = f"Unable to retrieve {key_path} metadata"
+        LOGGER.error(message)
+        raise ex(message)
```

Comment:
This threw an error when your test code triggered it. ex is non-callable.

Reply:
I got the exception when running from my side. maybe retry and see.

Reply:
The error we saw was raised from an un-mocked get_dbconnect_info, so it is an accurate reflection of a real error.

Reply:
current I dont have any aws secretsmanager secrets in my free tier aws account so it was giving an error of "no GetSecret". 
I think if you run it in the NGAP aws account, it should work

Reply:
Right, but if an error IS thrown, then this error handling code will fail.

Reply:
if an error is thrown, it will raise an exception.

except Exception as ex:        
message = f"Unable to retrieve {key_path} metadata"        
LOGGER.error(message)        
raise ex(message)

Reply:
Nope. Run this code.

try:
    raise Exception("blah")
except Exception as ex:
    raise ex("hello")

You will get the error
TypeError: 'Exception' object is not callable

Reply:
ok that makes sense. yes you are right. Changed to the following: 

try:
raise Exception("blah")
except Exception:
raise Exception("hello")